ptp/c4u/clock/clock.go

```diff
@@ -49,44 +59,76 @@ func (rb *RingBuffer) Write(c *ptp.ClockQuality) {
 }
 
 // Export data from the ring buffer
-func (rb *RingBuffer) Data() []*ptp.ClockQuality {
+func (rb *RingBuffer) Data() []*DataPoint {
 	return rb.data
 }
 
-func Worst(clocks []*ptp.ClockQuality) *ptp.ClockQuality {
+func Worst(points []*DataPoint, accuracyExpr string) (*ptp.ClockQuality, error) {
+	expr, err := prepareExpression(accuracyExpr)
+	if err != nil {
+		return nil, fmt.Errorf("evaluating accuracy math: %w", err)
+	}
+	offsets := []float64{}
 	var w *ptp.ClockQuality
-	for _, c := range clocks {
+	for _, c := range points {
 		if c == nil {
 			continue
 		}
-
 		if w == nil {
 			w = &ptp.ClockQuality{}
 		}
-
+		offsets = append(offsets, float64(c.PHCOffset))
+		// get data from oscillatord status
+		cq := clockQualityFromOscillatord(c.OscillatordStatus)
 		// Higher value of accuracy means worse
-		if c.ClockAccuracy > w.ClockAccuracy {
-			w.ClockAccuracy = c.ClockAccuracy
+		if cq.ClockAccuracy > w.ClockAccuracy {
+			w.ClockAccuracy = cq.ClockAccuracy
 		}
 
 		// Assuming higher class means worse
-		if c.ClockClass > w.ClockClass {
-			w.ClockClass = c.ClockClass
+		if cq.ClockClass > w.ClockClass {
+			w.ClockClass = cq.ClockClass
```

Comment:
It's not an immediate problem but we will have to come up with a better logic for ClockClass evaluation as well. It doesn't flap that often but having p99 (or even a variance) is useful here as well.

Reply:
Maybe in a next PR? :-) And how do you see - configurable or not? If configurable - how do we make sure the result is a valid clock class?

Reply:
Next PR sounds fine. What should be configurable? Formula? I am not fully convinced we need to have it configurable for ClockAccuracy even :)

Reply:
Yep, formula. Me neither, but it's so much easier to tune it while it's configurable at this stage.